### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by shloch**

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        for i in 0...n-1
+           #block
+            var = yield arr[i], arr[i+1]
+            if var > 0
+               arr[i+1], arr[i] = arr[i], arr[i+1]
+            end
+        end
+        n = n-1
+    end
+    puts arr
```
Comment (zlayabekrija) [inline bubble_sort.rb]: please remove puts from code

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    while (n > 1) do
+        for i in 0...n-1 
+            if arr[i] > arr[i+1]
+                arr[i+1], arr[i] = arr[i], arr[i+1]
+            end 
+        end 
+        n = n-1
+    end
+
+    print arr.to_s + "\n"
```
Comment (zlayabekrija) [inline bubble_sort.rb]: Please remove print,puts or p from code.

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+
+def bubble_sort_by(arr)
+
+   #puts 'yo'
+    n = arr.length
+    while (n > 1) do
+        for i in 0...n-1
+           #block
+            var = yield arr[i], arr[i+1]
```
Comment (zlayabekrija) [inline bubble_sort.rb]: what if I forget to provide you with a block. Please take this into consideration.

**Review by zlayabekrija — CHANGES_REQUESTED** [review]
Excellent work, please fix these minor things and I would like to suggest to implement this sorting algorithm in Enumerable method.
Please resubmit the new review request.
Happy coding. 😄

**Comment by shloch** [comment]
Thanks for the review.....i submitted proposed solution

File: bubble_sort.rb [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+=end
+
+##========== USING YIELD =======================================
+def bubble_sort_by(arr)
+
+    n = arr.length
+    
+    while (n > 1) do
+        for i in 0...n-1
+            var = yield arr[i], arr[i+1]
```
Comment (zlayabekrija) [inline bubble_sort.rb]: before checking for yield, first check if there is any block given

**Review by zlayabekrija — CHANGES_REQUESTED** [review]
You are one line from completing please follow up, with suggestion and resubmit review request.
Happy coding. 😄

File: bubble_sort.rb [hunk 5]
```diff
@@ -76,9 +78,13 @@ def bubble_sort_by(arr)
...
-            var = yield arr[i], arr[i+1]
-            if var > 0
-               arr[i+1], arr[i] = arr[i], arr[i+1]
+            if block_given?
+                var = yield arr[i], arr[i+1]
+                if var > 0
+                arr[i+1], arr[i] = arr[i], arr[i+1]
+                end
+            else
+                "no block"
```
Comment (zlayabekrija) [inline bubble_sort.rb]: I the case without a block, you should do the same as in regular bubble sort, sort it in order.
` if arr[i] > arr[i+1]
                arr[i+1], arr[i] = arr[i], arr[i+1]
end `, and with this you have cover both scenarions in bubble_sort_by

**Review by zlayabekrija — CHANGES_REQUESTED** [review]
Please fix the commented line, and resubmit review request.
Happy coding.

File: bubble_sort.rb (hunk 5 above)
Comment (shloch) [reply bubble_sort.rb]: thanks for your reviews

i added the corrections

**Review by zlayabekrija — APPROVED** [review]
Great work, good luck with the next projects. Thumbs up 👍  Happy Coding.

**Pull request merged**